apps/x86/halide/unsharp/unsharp.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+
+from exo import *
+from exo.libs.memories import DRAM_STACK
+from exo.platforms.x86 import *
+
+from exo.stdlib.scheduling import *
+from exo.stdlib.halide_scheduling_ops import *
+from exo.stdlib.inspection import get_enclosing_loop_by_name, get_enclosing_loop
+from exo.stdlib.stdlib import vectorize
+
+
+avx_f32_insts = [
+    mm256_loadu_ps,
+    mm256_storeu_ps,
+    mm256_mul_ps,
+    mm256_add_ps,
+    mm256_sub_ps,
+    mm256_div_ps,
+    mm256_broadcast_ss,
+    mm256_prefix_load_ps,
+    mm256_prefix_store_ps,
+    mm256_prefix_add_ps,
+    mm256_prefix_mul_ps,
+    mm256_prefix_sub_ps,
+    mm256_prefix_div_ps,
+    mm256_prefix_broadcast_ss,
+]
+
+
+def halide_vectorize(p, buffer: str, loop: str, width: int):
+    loop = get_enclosing_loop_by_name(p, p.find(f"{buffer} = _"), loop)
+    p = vectorize(p, loop, width, "f32", AVX2, avx_f32_insts, rules=[])
+
+    return p
+
+
+# from math import exp, pi, sqrt
+# sigma = 1.5
+# kernel = [exp(-x * x / (2 * sigma * sigma)) / (sqrt(2 * pi) * sigma) for x in range(4)]
```

Comment:
What if you defined this as a:
k0, k1, k2, k3 = kernel

Then, below assign k0, k1, k2, k3 to kernel[0] kernel[1] kernel[2] kernel[3].